server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestListTasksAction.java

```diff
@@ -103,18 +103,17 @@ public RestResponse buildResponse(T response, XContentBuilder builder) throws Ex
                     return new BytesRestResponse(RestStatus.OK, builder);
                 }
             };
-        } else if ("none".equals(groupBy)) {
+        } else if ("parents".equals(groupBy)) {
```

Comment:
Parser for `ListTasksResponse` uses only `groupby=node` response format, so better `toXContent` also serialize to `groupby=node` format. It's not necessarily but a nice thing as it seems for me.

Reply:
So this changes the structure of the Rest response output? In this case we shouldn't do it for backward compatiblity reasons. This change should also be merged to the 6.x branches I think.

Reply:
Yes, for the requests coming though the client I see this, what about requests that set `groupby` differently themselves? I see you also changes ListTaskResponse#toXContent to use toXContentGroupedByNone instead of toXContentGroupedByParents, I think thats why this switch here doesn't change anything but I'd still like to understand the reason for that change better.

Reply:
I am sorry, I thought all this conversation was about not only this 2 lines of code switching 2 parameters in `if` clause, but about all the changes including `ListTaskResponse#toXContent` to use `toXContentGroupedByNone` instead of `toXContentGroupedByParents`.
So all these changes doesn't change REST endpoint behavior. It just change `toXContent` result to be `groupby=none`, it's not necessarily but can bring:
- Idiomatically correct serialization/deserialization workflow: `toXContent` returns content which can be deserialized in `fromXContent`
- Tests can use Serialization handlers like `XContentHelper::toXContent` or can be fully automated i.e. extend `AbstractStreamableXContentTestCase` or `AbstractSerializingTestCase` (Currently not applicable because of broken `equals`)

And again it just a cosmetic change and I am happy to revert it back :)